content/en/blog/_posts/2020-03-30-topology-manager-beta.md

```diff
@@ -0,0 +1,513 @@
+---
+layout: blog
+title: "Kubernetes Topology Manager Moves to Beta - Align Up!"
+date: 2020-03-30
+slug: kubernetes-1-18-feature-topoloy-manager-beta
+---
+
+**Authors:** Kevin Klues (NVIDIA), Victor Pickard (RedHat), Conor Nolan (Intel)
+
+This blog post describes the **<code>TopologyManager</code>**, a beta feature of Kubernetes in release 1.18. The <strong><code>TopologyManager</code></strong> feature enables NUMA alignment of CPUs and peripheral devices (such as SR-IOV VFs and GPUs), allowing your workload to run in an environment optimized for low-latency.
+
+Prior to the introduction of the **<code>TopologyManager</code>**, the CPU and Device Manager would make resource allocation decisions independent of each other. This could result in undesirable allocations on multi-socket systems, causing degraded performance on latency critical applications. With the introduction of the <strong><code>TopologyManager</code></strong>, we now have a way to avoid this.
+
+This blog post covers:
+1. A brief introduction to NUMA and why it is important
+2. The policies available to end-users to ensure NUMA alignment of CPUs and devices
+3. The internal details of how the **<code>TopologyManager</code>** works
+4. Current limitations of the <strong><code>TopologyManager</code></strong>
+5. Future directions of the <strong><code>TopologyManager</code></strong>
+
+## So, what is NUMA and why do I care?
+
+The term NUMA stands for Non-Uniform Memory Access. It is a technology available on multi-cpu systems that allows different CPUs to access different parts of memory at different speeds. Any memory directly connected to a CPU is considered “local” to that CPU and can be accessed very fast. Any memory not directly connected to a CPU is considered “non-local” and will have variable access times depending on how many interconnects must be passed through in order to reach it. On modern systems, the idea of having “local” vs. “non-local” memory can also be extended to peripheral devices such as NICs or GPUs. For high performance, CPUs and devices should be allocated such that they have access to the same local memory.
+
+All memory on a NUMA system is divided into a set of “NUMA nodes”, with each node representing the local memory for a set of CPUs or devices. We talk about an individual CPU as being part of a NUMA node if its local memory is associated with that NUMA node. 
+
+We talk about a peripheral device as being part of a NUMA node based on the shortest number of interconnects that must be passed through in order to reach it.
+
+For example, in Figure 1, CPUs 0-3 are said to be part of NUMA node 0, whereas CPUs 4-7 are part of NUMA node 1. Likewise GPU 0 and NIC 0 are said to be part of NUMA node 0 because they are attached to Socket 0, whose CPUs are all part of NUMA node 0. The same is true for GPU 1 and NIC 1 on NUMA node 1.
+
+
+![drawing](../../../../static/images/blog/2020-03-25-kubernetes-1.18-release-announcement/example-numa-system.png)
```

Comment:
```suggestion
![drawing](/images/blog/2020-03-25-kubernetes-1.18-release-announcement/example-numa-system.png)
```
Hugo image paths need to be relative to the static directory.

I would also change the directory they're in from release-announcement to: `2020-03-30-topology-manager-beta` -- but this could be a follow up thing as the goal is to get this out the door ASAP^^:;;

Reply:
Thanks, I have fixed this in the latest version